src/components/RangeSlider.react.js

```diff
@@ -58,14 +59,51 @@ export default class RangeSlider extends Component {
             tipProps = tooltip;
         }
 
+        const truncatedMarks = this.props.marks
+            ? pickBy(
+                  (k, mark) => mark >= this.props.min && mark <= this.props.max,
+                  this.props.marks
+              )
+            : this.props.marks;
+
+        let style = {
+            padding: '25px',
+        };
+
+        if (
+            !vertical &&
+            (!tooltip ||
+                !tooltip.always_visible ||
+                !contains(tooltip.placement, ['top', 'topLeft', 'topRight']))
+        ) {
+            style = merge(style, {paddingTop: '0px'});
+        }
+
+        if (
+            vertical &&
+            (!tooltip ||
+                !tooltip.always_visible ||
+                !contains(tooltip.placement, [
+                    'left',
+                    'topRight',
+                    'bottomRight',
+                ]))
+        ) {
+            style = merge(style, {paddingLeft: '0px'});
+        }
+
+        if (vertical) {
+            style = merge(style, {height: verticalHeight + 'px'});
+        }
```

Comment:
* Could be simplified with shared vertical/!vertical checks.
* Minor point, could be memoized `(style, vertical, tooltip) -> style` to prevent style re-evaluation below